File: src/backend/src/types/s3.rs
use candid::CandidType;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, CandidType, Serialize, Deserialize)]
pub struct UploadFileData {
    pub filename: String,
    pub content_type: String,
    pub file_size: u64,
}

#[derive(Clone, Debug, CandidType, Serialize, Deserialize)]
pub struct S3Config {
    pub bucket_name: String,
    pub region: String,
    pub access_key_id: String,
    pub secret_access_key: String,
    pub endpoint: Option<String>,
}

File: src/frontend/src/types/s3.ts
export interface S3Config {
  bucket_name: string;
  region: string;
  access_key_id: string;
  secret_access_key: string;
  endpoint?: string;
}

export interface UploadFileData {
  filename: string;
  content_type: string;
  file_size: number;
}

export interface UploadResult {
  success: boolean;
  message: string;
  file_url?: string;
  file_id?: string;
}

export interface PhysicalArtSession {
  session_id: string;
  username: string;
  art_title: string;
  description: string;
  uploaded_photos: string[];
  created_at: bigint;
  updated_at: bigint;
  status: string;
}
